lib/puppet/pops/lookup/hiera_config.rb

```diff
@@ -249,16 +249,18 @@ def scope_interpolations_stable?(lookup_invocation)
     if @scope_interpolations.empty?
       true
     else
-      scope = lookup_invocation.scope
-      lookup_invocation.without_explain do
-        @scope_interpolations.all? do |key, root_key, segments, old_value|
-          value = scope[root_key]
-          unless value.nil? || segments.empty?
-            found = nil;
-            catch(:no_such_key) { found = sub_lookup(key, lookup_invocation, segments, value) }
-            value = found;
+      Puppet.override(avoid_hiera_lookup_errors: Puppet.lookup(:during_compilation){false}) do
+        scope = lookup_invocation.scope
+        lookup_invocation.without_explain do
+          @scope_interpolations.all? do |key, root_key, segments, old_value|
+            value = scope[root_key]
```

Comment:
I noticed that `Puppet::Parser::Scope#[]` accepts an options hash, e.g. `scope[root_key, options]`. So I wondered if it was possible to pass an argument like `lookup_invocation: true` (so it's more direct than using the puppet context). But I think I ran into the same issue you did when a manifest contains the `lookup` function and we want strict to raise an error.

Reply:
hmmm, yeah, we would still have to somehow pass in a flag from the `hiera_config` class, but it might be better isolated there than in the compiler.